Mage.Sets/src/mage/cards/k/KaylasMusicBox.java

```diff
@@ -0,0 +1,176 @@
+        MageObject sourceObject = source.getSourceObject(game);
+        if (controller == null || sourceObject == null) {
+            return false;
+        }
+
+        Card card = controller.getLibrary().getFromTop(game);
+        if (card == null) {
+            return false;
+        }
+
+        card.setFaceDown(true, game);
+        controller.lookAtCards(null, card, game);
+
+        if (controller.moveCardsToExile(card, source, game, true, CardUtil.getExileZoneId(game, source), CardUtil.getSourceName(game, source))) {
+            card.setFaceDown(true, game);
+            // No other player may look at the face-down cards you own exiled with Kayla’s Music Box, even if another player takes control of it. 
+            // (2022-10-14)
+            ContinuousEffect effect = new KaylasMusicBoxLookEffect(controller.getId());
+            effect.setTargetPointer(new FixedTarget(card.getId(), game));
+            game.addEffect(effect, source);
+        }
+        return true;
+    }
+
+}
+
+class KaylasMusicBoxLookEffect extends AsThoughEffectImpl {
+
+    private final UUID authorizedPlayerId;
+
+    public KaylasMusicBoxLookEffect(UUID authorizedPlayerId) {
+        super(AsThoughEffectType.LOOK_AT_FACE_DOWN, Duration.EndOfGame, Outcome.Benefit);
+        this.authorizedPlayerId = authorizedPlayerId;
+        staticText = "You may look at the cards exiled with {this}";
+    }
+
+    private KaylasMusicBoxLookEffect(final KaylasMusicBoxLookEffect effect) {
+        super(effect);
+        this.authorizedPlayerId = effect.authorizedPlayerId;
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public KaylasMusicBoxLookEffect copy() {
+        return new KaylasMusicBoxLookEffect(this);
+    }
+
+    @Override
+    public boolean applies(UUID objectId, Ability source, UUID affectedControllerId, Game game) {
+        UUID cardId = getTargetPointer().getFirst(game, source);
+        if (cardId == null) {
+            this.discard(); // card is no longer in the origin zone, effect can be discarded
+        }
+        return affectedControllerId.equals(authorizedPlayerId)
+                && objectId.equals(cardId);
+    }
+
+}
+
+class KaylasMusicBoxPlayFromExileEffect extends AsThoughEffectImpl {
+
+    KaylasMusicBoxPlayFromExileEffect() {
+        super(AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, Duration.EndOfTurn, Outcome.Benefit);
+        staticText = "until end of turn, you may play cards you own exiled with {this}.";
+    }
+
+    private KaylasMusicBoxPlayFromExileEffect(final KaylasMusicBoxPlayFromExileEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public KaylasMusicBoxPlayFromExileEffect copy() {
+        return new KaylasMusicBoxPlayFromExileEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public void init(Ability source, Game game) {
```

Comment:
no need to override a method just to call the super (unless there's an explanatory comment)